### PR TITLE
3.15.1 version OHOS adaptation

### PR DESCRIPTION
**Pull request by wanghui187**

### CODE REVIEW

File: cmake/Modules/CocosBuildHelpers.cmake [hunk 1]
```diff
@@ -139,6 +140,8 @@ elseif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     set(APPLE TRUE)
     set(SYSTEM_STRING "Mac OSX")
   endif()
+elseif(OHOS)
+  set(SYSTEM_STRING "OHOS")
```
Comment (minggo) [inline cmake/Modules/CocosBuildHelpers.cmake]: Should change to `HarmonyOS Next`?

File: cmake/Modules/FindCURL.cmake [hunk 2]
```diff
@@ -58,6 +58,25 @@ if(NOT CURL_FOUND)
       # Windows older "Win32 - MSVC" prebuilts (libcurl.lib, e.g. libcurl-7.15.5-win32-msvc.zip):
         libcurl
     )
+
+    if(OHOS)
+        # 主动set路径
```
Comment (minggo) [inline cmake/Modules/FindCURL.cmake]: Please don't use chinese.

File: cmake/Modules/FindChipmunk.cmake [hunk 3]
```diff
@@ -60,6 +60,24 @@ FIND_LIBRARY(CHIPMUNK_LIBRARY
 
 set(CHIPMUNK_INCLUDE_DIRS "${CHIPMUNK_INCLUDE_DIR}")
 
+if(OHOS)
+  # 主动set路径
```
Comment (minggo) [inline cmake/Modules/FindChipmunk.cmake]: Please don't use Chinese.

File: cmake/Modules/FindFreetype.cmake [hunk 4]
```diff
@@ -120,6 +120,30 @@ find_library(FREETYPE_LIBRARY
     [HKEY_LOCAL_MACHINE\\SOFTWARE\\gtkmm\\2.4;Path]
 )
 
+if(OHOS)
+    # 主动set路径
```
Comment (minggo) [inline cmake/Modules/FindFreetype.cmake]: Please don't use Chinese.

Comment (minggo) [reply cmake/Modules/FindFreetype.cmake]: Please fix it in other files.

File: cocos/scripting/js-bindings/manual/ScriptingCore.cpp [hunk 5]
```diff
@@ -337,6 +337,8 @@ bool JSBCore_os(JSContext *cx, uint32_t argc, jsval *vp)
     os = JS_InternString(cx, "OS X");
 #elif (CC_TARGET_PLATFORM == CC_PLATFORM_WINRT)
     os = JS_InternString(cx, "WINRT");
+#elif (CC_TARGET_PLATFORM == CC_PLATFORM_OHOS)
+    os = JS_InternString(cx, "OHOS");
```
Comment (minggo) [inline cocos/scripting/js-bindings/manual/ScriptingCore.cpp]: Should change to `HarmonyOS Next`?

File: cocos/scripting/js-bindings/script/jsb_boot.js [hunk 6]
```diff
@@ -1166,6 +1166,13 @@ var _initSys = function () {
      * @type {string}
      */
     sys.OS_WINRT = "WINRT";
+    /**
+     * @memberof cc.sys
+     * @name OS_OH
+     * @constant
+     * @type {string}
+     */
+    sys.OS_OHOS = "OHOS";
```
Comment (minggo) [inline cocos/scripting/js-bindings/script/jsb_boot.js]: Should change to `HarmonyOS Next`?

File: cocos/scripting/lua-bindings/CMakeLists.txt [hunk 7]
```diff
@@ -123,12 +156,39 @@ if(MACOSX)
...
 
 set(lua_bindings_files ${lua_cocos2d_source_files} ${lua_bindings_manual_files} ${lua_bindings_auto_files})
 
 add_library(luacocos2d ${lua_bindings_files})
 target_link_libraries(luacocos2d cocos2d)
+
+if(USE_LUAJIT)
+    target_link_libraries(luacocos2d luajit)
+    # luajit比lua多一个头文件luajit.h，在../../../external/lua/luajit/src下，暂时没有用到
+endif()
```
Comment (minggo) [inline cocos/scripting/lua-bindings/CMakeLists.txt:190]: Please don't use Chinese.

File: cocos/scripting/lua-bindings/script/framework/device.lua [hunk 8]
```diff
@@ -58,6 +58,8 @@ elseif target == cc.PLATFORM_OS_WINRT then
     device.platform = "winrt"
 elseif target == cc.PLATFORM_OS_WP8 then
     device.platform = "wp8"
+elseif target == cc.PLATFORM_OS_OHOS then
+    device.platform = "harmony"
```
Comment (minggo) [inline cocos/scripting/lua-bindings/script/framework/device.lua]: Should change to `HarmonyOS Next`?

File: cocos/base/CCEventListenerController.cpp [hunk 9]
```diff
@@ -66,6 +66,8 @@ bool EventListenerController::init()
             break;
         case EventController::ControllerEventType::BUTTON_STATUS_CHANGED:
             {
+#if (CC_TARGET_PLATFORM != CC_PLATFORM_WINRT)
```
Comment (minggo) [inline cocos/base/CCEventListenerController.cpp]: Is it a mistake?

File: cmake/Modules/CocosBuildHelpers.cmake (hunk 1 above)
Comment (wanghui187) [reply cmake/Modules/CocosBuildHelpers.cmake]: Fixed

File: cmake/Modules/FindCURL.cmake (hunk 2 above)
Comment (wanghui187) [reply cmake/Modules/FindCURL.cmake]: Fixed

File: cmake/Modules/FindChipmunk.cmake (hunk 3 above)
Comment (wanghui187) [reply cmake/Modules/FindChipmunk.cmake]: Fixed

File: cmake/Modules/FindFreetype.cmake (hunk 4 above)
Comment (wanghui187) [reply cmake/Modules/FindFreetype.cmake]: Fixed

File: cocos/scripting/js-bindings/manual/ScriptingCore.cpp (hunk 5 above)
Comment (wanghui187) [reply cocos/scripting/js-bindings/manual/ScriptingCore.cpp]: Fixed

File: cocos/scripting/js-bindings/script/jsb_boot.js (hunk 6 above)
Comment (wanghui187) [reply cocos/scripting/js-bindings/script/jsb_boot.js]: Fixed

**Pull request merged**